eng/pipelines/templates/stages/archetype-sdk-client.yml

```diff
@@ -23,23 +23,40 @@ parameters:
 - name: TargetDocRepoName
   type: string
   default: azure-docs-sdk-python
-- name: SkipPythonVersion
-  type: string
-  default: ''
-- name: AdditionalTestMatrix
-  type: object
-  default: []
 - name: DevFeedName
   type: string
   default: public/azure-sdk-for-python
 - name: BuildTargetingString
   type: string
   default: azure-*
+- name: MatrixConfigs
+  type: object
+  default:
+    - Name: Python_ci_test_base
+      Path: eng/pipelines/templates/stages/platform-matrix.json
+      Selection: sparse
+      GenerateVMJobs: true
+- name: PRMatrixConfigs
```

Comment:
The entire reason they had the PRMatrix configs before was to have a way to create a sparse matrix for PR validation. Given we now can do that with the new format do we really need this?

Reply:
The current setup isn't a sparse matrix as we treat it, because sparse matrix means "make sure every parameter in the config appears at least once" whereas the behavior in master is to skip testing half the python versions.

Reply:
I still kind of feel this pattern is going against what we have in the other languages. If we feel we want a different matrix for PR vs not we should make that a common pattern across the languages. Currently if we can avoid it I'd like to not have a unique configuration for PR but instead just use a sparse version of the full matrix.

Reply:
I think this approach represents a sparse version of the full matrix achieved with the tools we have available (sparseness operating at the parameter key boundary). The alternatives I can see are defining an independent matrix, defining matrix filter defaults in the yaml, or expanding the version list to use groupings containing a `RunForPR` variable. I think having a generic way to say "pick X sparse subset of Y key" is going to be problematic, because in most scenarios there will be a priority list of which X values are most important (whereas the current sparseness always gives an "everything is covered" guarantee). This method is essentially "pick X values from a shared base config."

Another alternative is to remove the PR vs. master split and just run all the tests on PR.

Reply:
All that said, we could introduce a `limit` option to the sparse generator, which would extract N elements from an already generated sparse matrix (with indexes spread out across the length of that array).